packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -165,6 +180,27 @@ export function getChildHostContext(
   return getChildNamespace(parentNamespace, type);
 }
 
+export function getChildHostContextForEvent(
+  parentHostContext: HostContext,
+  event: ReactEventComponent | ReactEventTarget,
+): HostContext {
+  if (__DEV__) {
+    const parentHostContextDev = ((parentHostContext: any): HostContextDev);
+    if (event.$$typeof === REACT_EVENT_COMPONENT_TYPE) {
+      return {...parentHostContextDev, isEventComponent: true};
+    } else if (event.$$typeof === REACT_EVENT_TARGET_TYPE) {
+      warning(
+        typeof parentHostContext === 'function' &&
+          parentHostContext !== null &&
+          parentHostContext.$$typeof === REACT_EVENT_COMPONENT_TYPE,
+        'validateDOMNesting: React event targets must be direct children of event components.',
+      );
+      return {...parentHostContextDev, isEventTarget: true, hostNodeCount: 0};
```

Comment:
Let's explicitly list out the props you want to copy here so it's clear what overrides are expected and what is expected to reset.